### PR TITLE
DOC: Better docs for `jnp.fft.hfft` and `jnp.fft.ihfft`

### PR DESCRIPTION
**Pull request by rajasekharporeddy**
Part of #21461

### CODE REVIEW

File: jax/_src/numpy/fft.py [hunk 1]
```diff
@@ -522,18 +522,131 @@ def irfft(a: ArrayLike, n: int | None = None,
...
+
+  See also:
+    - :func:`jax.numpy.fft.ihfft`: Computes a one-dimensional inverse FFT of an
+      array whose spectrum has Hermitian symmetry.
+    - :func:`jax.numpy.fft.fft`: Computes a one-dimensional discrete Fourier
+      transform.
+    - :func:`jax.numpy.fft.rfft`: Computes a one-dimensional discrete Fourier
+      transform of a real-valued input.
+
+  Examples:
```
Comment (dfm) [inline jax/_src/numpy/fft.py:555]: Could we add an example where the input data are complex? I think we could compute an appropriate input using `rfft`?

File: jax/_src/numpy/fft.py [hunk 2]
```diff
@@ -522,18 +522,131 @@ def irfft(a: ArrayLike, n: int | None = None,
...
+      along ``axis``.
+    axis: optional, int, default=-1. Specifies the axis along which the transform
+      is computed. If not specified, the transform is computed along axis -1.
+    norm: optional, string. The normalization mode. "backward", "ortho" and "forward"
+      are supported. Default is "backward".
+
+  Returns:
+    An array containing the one-dimensional discret Fourier transform of ``a``
+    by exploiting its inherent Hermitian-symmetry, having a dimension of ``n``
+    along ``axis``.
```
Comment (dfm) [inline jax/_src/numpy/fft.py:545]: Is it worth stating that the output will always be real?

File: jax/_src/numpy/fft.py [hunk 3]
```diff
@@ -522,18 +522,131 @@ def irfft(a: ArrayLike, n: int | None = None,
...
+      transform of a real-valued input.
+
+  Examples:
+    >>> x = jnp.array([[1, 3, 5, 7],
+    ...                [2, 4, 6, 8]])
+    >>> jnp.fft.hfft(x)
+    Array([[24., -8.,  0., -2.,  0., -8.],
+           [30., -8.,  0., -2.,  0., -8.]], dtype=float32)
+
+    This is equivalent to computing ``jnp.fft.fft`` of the following array ``x1``:
```
Comment (dfm) [inline jax/_src/numpy/fft.py:562]: Not quite equivalent because of the `dtype`s!

**Review by dfm — CHANGES_REQUESTED** [review]
Thanks! Small comments inline.

File: jax/_src/numpy/fft.py (hunk 1 above)
Comment (rajasekharporeddy) [reply jax/_src/numpy/fft.py:555]: Thanks. Included the example with complex input.

File: jax/_src/numpy/fft.py (hunk 2 above)
Comment (rajasekharporeddy) [reply jax/_src/numpy/fft.py]: Written it as "A real-vlaued array"

File: jax/_src/numpy/fft.py (hunk 3 above)
Comment (rajasekharporeddy) [reply jax/_src/numpy/fft.py]: Modified the statement as "This value is equal to the real component of the discrete Fourier transform
    of the following array ``x1`` computed using ``jnp.fft.fft``."

**Comment by dfm** [comment]
Looks good - thanks! Can you squash the commits?

**Comment by rajasekharporeddy** [comment]
Thanks! Squashed into single commit

**Review by dfm — APPROVED** [review]
Thanks!

**Pull request merged**